examples/admission_controller.rs

```diff
@@ -0,0 +1,93 @@
+use std::convert::{Infallible, TryInto};
+
+use kube::api::{AdmissionRequest, AdmissionResponse, AdmissionReview, DynamicObject, PatchType};
+#[macro_use] extern crate log;
+use warp::{reply, Filter, Reply};
+
+#[tokio::main]
+async fn main() {
+    std::env::set_var("RUST_LOG", "debug,kube=debug");
+    tracing_subscriber::fmt::init();
+
+    let routes = warp::path("mutate")
+        .and(warp::body::json())
+        .and_then(mutate)
+        .with(warp::trace::request());
+
+
+    // For admission controllers running behind a service, you must generate a
+    // certificate for the service and encode the CA in the
+    // MutatingWebhookConfiguration, then terminate TLS here. For controllers
+    // running outside of the cluster, while still necessary, it's fine to
+    // handle TLS upstream. See:
+    // https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/#tls-certificates
+
+    // warp::serve(warp::post().and(routes))
+    //     .tls()
+    //     .cert_path("admission-controller-tls.crt")
+    //     .key_path("admission-controller-tls.key")
+    //     .run(([0, 0, 0, 0], 8443))
+    //     .await;
+
+    warp::serve(warp::post().and(routes))
+        .run(([0, 0, 0, 0], 8080))
+        .await;
+}
+
+async fn mutate(body: AdmissionReview<DynamicObject>) -> Result<impl Reply, Infallible> {
+    let mut res: AdmissionResponse = Default::default();
```

Comment:
I think we could potentially convert `From` Review to a Response as well, to ensure people always copy the `uid`.

Reply:
I think all we need is `Serialize + Deserialize` as the bounds, but I sorta wanted to make it clear the only thing that could be there is a `Resource` even if we use none of the trait.